src/node.cc

```diff
@@ -3833,6 +4200,13 @@ static void ParseArgs(int* argc,
 
   // Copy remaining arguments.
   const unsigned int args_left = nargs - index;
+
+  if (is_env && args_left) {
+    fprintf(stderr, "%s: non-option '%s' is not allowed in NODE_OPTIONS\n",
```

Comment:
The error message here is not very clear. Perhaps something like `%s is not supported in NODE_OPTIONS\n`